ethpm_types/utils.py

```diff
@@ -1,4 +1,40 @@
 from hashlib import md5
+from typing import Type
+
+from eth_utils import to_bytes, to_hex
+
+
+class Bytes32(bytes):
+    """
+    Custom bytes wrapper class for working with pydantic models.
+    Converts str to bytes with a length of 32 and serializes as hex.
+    Inspired by HexBytes https://github.com/ethereum/hexbytes/blob/master/hexbytes/main.py
+    """
+
+    bytes_length: int = 32
+
+    def __new__(cls: Type[bytes], val: str):
+        val = to_bytes(hexstr=val)
+        if len(val) != Bytes32.bytes_length:
```

Comment:
Using it from the class like this won't work, I don't think

Reply:
I added the `new` method  to support not using within a pydantic model, it isn't required otherwise. This works best with the pydantic `BaseModel`. In my script testing, I was not able to access  class attributes on `cls`, but by explicitly calling `Bytes32`, I was able to access the attributes.

 I don't like this solution but wanted to put something up to get eyes on it.

Reply:
It probably only works for the default but won't work if it is set